qdrant-landing/content/articles/serverless.md

```diff
@@ -0,0 +1,290 @@
+
+```rust
+use anyhow::Result;
+use qdrant_client::prelude::*;
+use std::collections::HashMap;
+
+fn setup<'i>(
+    embed_client: &reqwest::Client,
+    embed_api_key: &str,
+    qdrant_url: &str,
+    api_key: Option<&str>,
+    collection_name: &str,
+    data: impl Iterator<Item = (&'i str, HashMap<String, Value>)>,
+) -> Result<()> {
+    let mut config = QdrantClientConfig::from_url(qdrant_url);
+    config.api_key = api_key;
+    let client = QdrantClient::new(Some(config))?;
+
+    // create the collections
+    if !client.has_collection(COLLECTION_NAME).await? {
+        client
+            .create_collection(&CreateCollection {
+                collection_name: collection_name.into(),
+                vectors_config: Some(VectorsConfig {
+                    config: Some(Config::Params(VectorParams {
+                        size: 1536, // our output dimensions from above
+                        distance: Distance::Cosine as i32,
+                        ..Default::default()
+                    })),
+                }),
+                ..Default::default()
+            })
+            .await?;
+    }
+    let mut id_counter = 0_u64;
+    let points = data.map(|(text, payload)| {
+        let id = std::mem::replace(&mut id_counter, *id_counter + 1);
+        let vectors = Some(embed(embed_client, text, embed_api_key).unwrap());
+        PointStruct { id, vectors, payload }
+    }).collect();
+    client.upsert_points(collection_name, points, None).await?;
+    Ok(())
+}
+```
+
+Depending on whether you want to efficiently filter the data, you can also add some indexes. We leave this out for brevity, but have [example code](https://github.com/qdrant/examples/tree/master/lambda-search) containing this operation. Also this does not implement chunking (splitting the data to upsert in multiple requests).
+
+Add a suitable `main` method and you can run this code to insert the points (or just use the binary from the example).
+
+Now that you have the points inserted, you can search them by embedding:
+
+```rust
+use anyhow::Result;
+use qdrant_client::prelude::*;
+pub async fn search(
+    text: &str,
+    collection_name: String,
+    client: &Client,
+    api_key: &str,
+    qdrant: &QdrantClient,
+) -> Result<Vec<ScoredPoint>> {
+    Ok(qdrant.search_points(&SearchPoints {
+        collection_name,
+        limit: 5, // use what fits your use case here
+        with_payload: Some(true.into()),
+        vector: embed(client, text, api_key)?,
+        ..Default::default()
+    }).await?.result)
+}
+```
+
+And that's it. Obviously, you can also filter by adding a `filter: ...` field to the `SearchPoints`, and you will likely want to process the result further, but the example code already does that, so feel free to start from there in case you need this functionality.
```

Comment:
can't start the sentence with "and" its against literary rules......either remove or elaborate.